packages/cli/src/lib/setup/setupUpgrade.ts

```diff
@@ -335,35 +380,35 @@ function Upgrade(options) {
             );
         }
         // Get the url of io-package.json or direct the version
-        if (!repoUrl[adapter]) {
+        if (!sources[adapter]) {
             console.log(`Adapter "${adapter}" is not in the repository and cannot be updated.`);
         }
-        if (repoUrl[adapter].controller) {
+        if (sources[adapter].controller) {
             return console.log(
                 `Cannot update ${adapter} using this command. Please use "iobroker upgrade self" instead!`
             );
         }
 
-        let ioInstalled;
+        // TODO: not 100 % true but should be correct enough
+        let ioInstalled: Partial<ioBroker.AdapterObject>;
```

Comment:
This would be better, because you don't need the `ioInstalled.common!` everywhere below:
```suggestion
        let ioInstalled: Pick<ioBroker.AdapterObject, 'common'>;
```

Reply:
And again, `ioInstalled.common.version` should just be a constant :)